src/udev/udevd.c

```diff
@@ -404,6 +398,27 @@ static int worker_lock_block_device(sd_device *dev, int *ret_fd) {
         if (r < 0)
                 return log_device_debug_errno(dev, r, "Failed to get devname: %m");
 
+        *ret = val;
+        return 1;
```

Comment:
this function has successfuly exit paths (i.e. where 0 is returned), where *ret is not initialized. that's not in line with our coding style, which after all says all succesful exit paths from a function should initialize all return values. or in other words, you need to sprinkle `*ret = NULL` at before various occurences of `return 0`, above.